ambry-store/src/test/java/com/github/ambry/store/BlobStoreCompactorTest.java

```diff
@@ -2914,19 +2907,12 @@ private void testDuplicatePutsHelper() throws Exception {
     compactor.initialize(state.index);
 
     long duplicateCount = compactor.getTgtMetrics().compactionTargetIndexDuplicateOnNonRecoveryCount.getCount();
-    if (alwaysEnableTargetIndexDuplicateChecking) {
+    try {
       compactor.compact(details, bundleReadBuffer);
-      long duplicateCountAfterCompaction =
-          compactor.getTgtMetrics().compactionTargetIndexDuplicateOnNonRecoveryCount.getCount();
-      assertEquals(duplicateCountAfterCompaction - duplicateCount, 1);
-    } else {
-      try {
-        compactor.compact(details, bundleReadBuffer);
-        fail("Should fail with duplicate put");
-      } catch (StoreException e) {
-        assertEquals(StoreErrorCodes.Unknown_Error, e.getErrorCode());
-        assertTrue(e.getMessage().contains("duplicate PUT"));
-      }
+      fail("Should fail with duplicate put");
+    } catch (StoreException e) {
+      assertEquals(StoreErrorCodes.Unknown_Error, e.getErrorCode());
+      assertTrue(e.getMessage().contains("duplicate PUT"));
```

Comment:
Should we upgrade Junit to 5 and use `assertThrows`?

Reply:
upgrading to Junit 5 takes a long time. Junit 5 changes the assert interface to put message after the values. In Junit 4, we can do
```
Assert.assertEquals("Something is wrong", expectedValue, obtainedValue);
```
But in Junit 5, we have to do 
```
Assert.assertEquals(expectedValue, obtainedValue, "Something is wrong");
```
It will take a long to fix all of that.